examples/dgcnn_segmentation.py

```diff
@@ -86,11 +86,8 @@ def test(loader):
     for data in loader:
         data = data.to(device)
         pred = model(data).argmax(dim=1)
-
-        i, u = i_and_u(pred, data.y, loader.dataset.num_classes, data.batch)
-        iou = i.cpu().to(torch.float) / u.cpu().to(torch.float)
-        iou[torch.isnan(iou)] = 1
-
+        iou = jaccard_index(pred, data.y,
```

Comment:
@rusty1s so you say We shall implement the batch strategy here, in the example, right?

Reply:
Let me verify the output in a mini-batch scenario (with a dynamic number of points per example) and come back to you (probably on Monday). This is my only concern. Otherwise the PR looks good to me (thank you very much).

Reply:
What I mean by this is that I think we need to compute `jacard_index` with `reduction = None`, and then apply the reduction by ourselves, e.g., via `scatter_mean(iou, batch, dim=0).mean()`.

Reply:
sure, because otherwise we can use also the `nn.Module` metrics version with `update` and `compute` :rabbit:

Reply:
cc: @SkafteNicki

Reply:
Updated the metric computation in the examples, in which IoU is now computed for each example individually and averaged afterwards. I think this leads to the correct computation now (please confirm).

I think there exists no alternative way to avoid the for-loop at the moment.

Reply:
This leads to:
`You have set 50 number of classes which is different from predicted (32) and target (3) number of classes`